### PR TITLE
[Merged by Bors] - chore(Tactic): clean up `q()` notation

### PR DESCRIPTION
**Pull request by eric-wieser**
There is no need to write `(q(some_lemma))`, `q(some_lemma)` already has the right precedence. This also removes some `by exact`s that were either cargo-culted, or fixed by a recent change to Lean / Quote4.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Order/Field/Power.lean [hunk 1]
```diff
@@ -254,7 +254,7 @@ def evalZPow : PositivityExt where eval {u α} zα pα e := do
       have m : Q(ℕ) := mkRawNatLit (n / 2)
       haveI' : $b =Q $m + $m := ⟨⟩ -- b = bit0 m
       haveI' : $e =Q $a ^ $b := ⟨⟩
-      pure (by exact .nonnegative q(zpow_bit0_nonneg $a $m))
+      pure (.nonnegative q(zpow_bit0_nonneg $a $m))
```
Comment (YaelDillies) [inline Mathlib/Algebra/Order/Field/Power.lean:257]: You could eliminate one more pair of brackets like so:
```suggestion
      return .nonnegative q(zpow_bit0_nonneg $a $m)
```

Comment (eric-wieser) [reply Mathlib/Algebra/Order/Field/Power.lean:257]: `return` isn't safe after `~q()` matching due to https://github.com/leanprover-community/quote4/issues/21, so I'd prefer not to risk breaking random things here.

**Review by YaelDillies — APPROVED** [review]
maintainer merge

File: Mathlib/Algebra/Order/Field/Power.lean (hunk 1 above)
Comment (ericrbg) [reply Mathlib/Algebra/Order/Field/Power.lean:257]: I wasn't aware that `return` is different to `pure`...

**Review by Vierkantor — APPROVED** [review]
Thanks!

bors merge

**Pull request closed without merging**